UnityProject/Assets/Scripts/Items/Medical/Gauze.cs

```diff
@@ -31,47 +31,41 @@ public override void ServerPerformInteraction(HandApply interaction)
 			}
 		}
 
-		private void RemoveLimbExternalBleeding(LivingHealthMasterBase targetBodyPart, HandApply interaction)
+		private void RemoveLimbExternalBleeding(LivingHealthMasterBase livingHealth, HandApply interaction)
 		{
-			foreach(var container in targetBodyPart.RootBodyPartContainers)
+			foreach(var bodyPart in livingHealth.BodyPartList)
 			{
-				if(container.BodyPartType == interaction.TargetBodyPart)
+				if(bodyPart.BodyPartType == interaction.TargetBodyPart)
 				{
-					foreach(BodyPart limb in container.ContainsLimbs)
+					if(bodyPart.IsBleedingExternally)
 					{
-						if(limb.IsBleedingExternally)
+						bodyPart.StopExternalBleeding();
+						if(interaction.Performer.Player().GameObject == interaction.TargetObject.Player().GameObject)
```

Comment:
the target object seems to be a body part, not the person

Reply:
Not sure I understand your comment. `interaction.Performer` is the player's gameobject, and the Player() extension returns `ConnectedPlayer`. The GameObject member is a reference to the `ConnectedPlayer()`'s gameobject, so it could just be shortened to `interaction.Performer`.

Reply:
targetobject is a bodypart, and thats a different network object, so player() doesnt get the connection if you use it, I guess that's the reason of why using gauzes on yourself doesnt trigger the proper message

Reply:
Ooh. I getcha. Thanks.